.CMake/compiler_opts.cmake

```diff
@@ -85,7 +85,7 @@ if(CMAKE_C_COMPILER_ID MATCHES "Clang|GNU")
     add_compile_options(${OQS_OPT_FLAG})
 
     # If this is not a dist build we also need to set the OQS_USE_[EXTENSION] flags
-    if(NOT ${OQS_DIST_BUILD})
+    if(NOT ${OQS_DIST_BUILD} AND NOT CMAKE_CROSSCOMPILING)
```

Comment:
For my personal understanding: Why is this change needed?

Reply:
The `try_run()` command inside `gcc_clang_instrinsics.cmake`  doesn't properly use the provided compiler flags and therefore can't produce a working executable using some embedded toolchains (e.g. the Zephyr one, but also applies to other toolchains). In this case, CMake fails and compilation is impossible. 

After the `try_run()` call, all CPU features have been disabled either way for cross compilation. Hence, I disabled the execution of the file completely for cross compilation.